test/widgets/lightbox_test.dart

```diff
@@ -558,4 +601,51 @@ void main() {
       check(platform.position).equals(position);
     });
   });
-}
+
+  group('LightboxHero', () {
+    testWidgets('no hero animation occurs between different message list pages for same image', (tester) async {
+      final channel = eg.stream(streamId: eg.defaultStreamMessageStreamId);
+      final message = eg.streamMessage(stream: channel,
+        contentMarkdown: ContentExample.spoilerHeaderHasImage.html, topic: 'test topic');
+      await setupMessageListPage(tester, narrow: const CombinedFeedNarrow(),
+        streams: [channel], messages: [message]);
+
+      connection.prepare(json:
+        eg.newestGetMessagesResult(foundOldest: true, messages: [message]).toJson());
```

Comment:
nit: move this to the line just before `tester.tap`, because it prepares an API response for that